### PR TITLE
Story/quad15

### PR DESCRIPTION
**Pull request by gunpreetsingh**
Unarchive changes.

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java [hunk 1]
```diff
@@ -324,5 +323,15 @@ public boolean getDeactiveStatus(String workspaceId) throws QuadrigaStorageExcep
             return wsDto.getIsdeactivated();
         return false;
     }
+    @Transactional
+    public boolean getArchiveStatus(String workspaceId) throws QuadrigaStorageException 
+    {
+        WorkspaceDTO wsDto = workspaceDao.getWorkspaceDTO(workspaceId.trim());
+        if (wsDto != null)
+        {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workspace/ListWSManager.java:331]: Please read the code style guide (linked on the course book page) and reformat the code.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java [hunk 2]
```diff
@@ -51,106 +53,32 @@ protected void initBinder(WebDataBinder binder)
...
-		workspaceForm.setWorkspaceList(archiveWorkspaceList);
-		
-		model.getModelMap().put("workspaceform", workspaceForm);
-		model.getModelMap().put("wsprojectid",projectid);
-		model.getModelMap().put("success", 0);
-		return model;
-	}
+	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT,paramIndex = 1, 
+	userRole = {RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR}),
+	@ElementAccessPolicy(type=CheckedElementType.WORKSPACE,paramIndex=2,userRole={})})
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ArchiveWSController.java:69]: You probably want to add some user roles here. Only workspace admins should be allowed to do this.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.workspace.IArchiveWSManager;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+@Controller
+public class UnarchiveWSController {
+	
+	@Autowired
+	IArchiveWSManager archiveWSManager;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java:26]: should be private

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+     * @param principal
+     * @return String - URL of the form
+     * @throws QuadrigaStorageException
+     * @author Gunpreet Singh
+     * @throws QuadrigaAccessException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT,paramIndex = 1, 
+    		userRole = {RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_CONTRIBUTOR}),
+    		@ElementAccessPolicy(type=CheckedElementType.WORKSPACE,paramIndex=2,userRole={})})
+    		@RequestMapping(value = "auth/workbench/{workspaceid}/unarchiveSingleWorkspace", method = RequestMethod.GET)
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java:43]: specify userRoles; only workspace admins should be allowed to do this.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    		public ModelAndView unarchiveWorkspace(
+    				@RequestParam("projectid") String projectid,
+    				@PathVariable("workspaceid") String workspaceid,
+    				Principal principal, RedirectAttributes redirectAttributes) throws QuadrigaStorageException, QuadrigaAccessException
+    		{
+    			ModelAndView model = new ModelAndView("redirect:/auth/workbench/"+ projectid);
+    			archiveWSManager.unArchiveWorkspace(workspaceid, principal.getName());		
+    			redirectAttributes.addFlashAttribute("show_success_alert", true);
+    			redirectAttributes.addFlashAttribute("success_alert_msg","The workspace has been successfully unarchived.");
+    	        return model;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java:54]: what's up with the formatting? Please review the style guide.

File: Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp [hunk 6]
```diff
@@ -399,6 +394,9 @@
 								Workspace</font>
 
 						</c:if>
+						<c:if test="${isDeactivated == true}">
+
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp:398]: Why is here an if statement that doesn't do anything?

File: Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp [hunk 7]
```diff
@@ -525,7 +523,77 @@
...
+				<c:if test="${isArchived == true}">
+						<a href="#" onclick="confirmWorkspaceUnarchive()"> 
+							Unarchive Workspace
+						</a>	
+				</c:if>
+				<c:if test="${isDeactivated == false && isArchived == false}">
+					<a href="#" onclick="confirmWorkspaceArchive()"> 
+							Archive Workspace
+					</a>
+					<a href="#" onclick="return confirmWorkspaceDeactivation();">
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp:587]: I don't think you need this "return" here, do you?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/workspace/IListWSManager.java [hunk 8]
```diff
@@ -57,6 +57,7 @@ String getProjectIdFromWorkspaceId(String workspaceId)
 			throws QuadrigaStorageException;
 	
 	public boolean getDeactiveStatus(String workspaceId) throws QuadrigaStorageException;
+	public boolean getArchiveStatus(String workspaceId) throws QuadrigaStorageException;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/workspace/IListWSManager.java:60]: Javadoc missing :)

**Comment by jdamerow** [comment]
see comments

File: Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp (hunk 6 above)
Comment (gunpreetsingh) [inline Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp:398]: removed

File: Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp (hunk 7 above)
Comment (gunpreetsingh) [inline Quadriga/src/main/webapp/WEB-INF/views/workspace/workspacedetails.jsp:587]: nope

File: Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java (hunk 5 above)
Comment (gunpreetsingh) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/UnarchiveWSController.java:54]: formatted!

**Pull request closed without merging**